### DISCUSSION TITLE
How to make particles properly spawn in front of the player?

### DISCUSSION THREAD

**Discussion by kz-n** [Minestom/Minestom#1304] [Help] [answered]
The title is a bit misleading but im not sure how i could phrase it, the issue is more of my code than particles... I'm trying to shoot a sequence of particles where the player is looking, however it looks like this:
https://cdn.discordapp.com/attachments/928063179622334465/1003781095558479933/2022-08-01_23-46-54.mp4
(its going the wrong direction)

I have tried to invert it using asVec() and rotate() but that just makes a mess..

This is the code I have

```java
eventHandler.addListener(PlayerHandAnimationEvent.class, event -> {
            Pos pos = event.getPlayer().getPosition().add(event.getPlayer().getPosition().direction());
            Pos diff = event.getPlayer().getPosition().sub(pos);

            for (int i = 0; i < 20 ; i++) {
                diff = diff.add(diff);
                var fnl = pos.add(diff);
                ParticlePacket particlePacket = ParticleCreator.createParticlePacket(Particle.ANGRY_VILLAGER, fnl.x(), fnl.y(), fnl.z(), 0f, 0f, 0f, 1);
                event.getPlayer().sendPacket(particlePacket);
            }
        });
```

**Comment by FluxCapacitor2** [comment, marked as answer]
I'm not exactly sure why your code doesn't work, but this works fine for me:
```java
MinecraftServer.getGlobalEventHandler().addListener(PlayerHandAnimationEvent.class, event -> {
        for(int i = 0; i < 20; i++) {
            Pos diff = event.getPlayer().getPosition().direction().mul(i)
            Pos finalPos = event.getPlayer().getPosition().add(diff) // Add the player's facing direction (times the multiplier) to their current position
            ParticlePacket packet = ParticleCreator.createParticlePacket(Particle.ANGRY_VILLAGER, finalPos.x(), finalPos.y(), finalPos.z(), 0f, 0f, 0f, 1)
            event.getPlayer().sendPacket(packet)
        }
    }
```
Just a small modification to the way you add the player's facing direction. I am using the `Vec#mul` method to multiply the player's direction vector by a number between 0 and 20 for each iteration, and that vector is added to the player's position to get the location of the particle.

**Reply by kz-n** [reply]
I had actually found a separate solution, but this also works and is arguably cleaner, Thanks.

**Comment by emortaldev** [comment]
You just need to swap the pos and player positon here: `event.getPlayer().getPosition().sub(pos);`

**Reply by kz-n** [reply]
where?
